src/ScottPlot/Regression.cs

```diff
@@ -0,0 +1,104 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace ScottPlot
+{
+	public interface IRegressionLine {
+		double[] GetCoefficients();
+		double[] Residual();
+		Plot Draw(Plot plot);
+	}
+
+	public class LinearRegressionLine : IRegressionLine {
+		private double[] x;
+		private double[] y;
+		private double[] coefficients;
+
+		public LinearRegressionLine(double[] xAxis, double[] yAxis) {
+			if (xAxis.Length != yAxis.Length) {
+				throw new ArgumentException("Axes of mismatched length");
+			}
+
+			x = xAxis;
+			y = yAxis;
+			coefficients = GetCoefficients();
+		}
+
+		private double SetMean(double[] set) {
+			double total = 0;
+
+			foreach (double curr in set) {
+				total += curr;
+			}
+
+			return total / set.Length;
+		}
+
+		private double y_hat(double x) {//Value predicted by regression model
+			//double[] coefficients = GetCoefficients();
+
+			return coefficients[0] + coefficients[1] * x;
+		}
+
+		public double[] GetCoefficients() {//y_hat = a+bx
+			double a;
+			double b;
+
+			double xMean=SetMean(x);
+			double yMean=SetMean(y);
+
+			double sumXYResidual = 0;
+			double sumXSquareResidual = 0;
+
+			for (int i = 0; i < x.Length; i++)
+			{
+				sumXYResidual += (x[i] - xMean) * (y[i] - yMean);
+				sumXSquareResidual += Math.Pow((x[i] - xMean), 2);
```

Comment:
Multiplication is faster than Pow() for integer exponents because Pow()'s exponent argument gets converted to a double, then it has to calculate a floating point exponent.

Reply:
Compilers more smart than we thinked, not yet implemented, but wil be made soon:
https://github.com/dotnet/coreclr/pull/26552
Converts:
`Math.Pow(x, 2) to x*x`